core/src/services/s3/backend.rs

```diff
@@ -942,6 +942,12 @@ impl Access for S3Backend {
                 write: true,
                 write_can_empty: true,
                 write_can_multi: true,
+                // Only S3 Express One Zone storage class supports append.
+                write_can_append: self
+                    .core
+                    .default_storage_class
```

Comment:
Hi, checking with `default_storage_class` doesn't seem like a good idea to me. How about adding a new flag called `enable_write_with_append`? Users who confirm that they support `append` can enable this.